content/5_Plat/Convex_Hull.mdx

```diff
@@ -85,22 +84,217 @@ export const problems = {
+
+To start the creation of the convex hull, we choose 2 points. The pivot (first point), and the second point based on our initial sorting.
+
+Then, we maintain a stack containing the points of the convex hull, such that the set of points in the stack is convex. As a result, we only have to check the candidate point against the topmost 2 points.
+
+In order for the hull to be convex, every three consecutive points of the stack must contain a counterclockwise turn.
+
+Denote our stack as $\texttt{C}$, the top element of $\texttt{C}$ as $\texttt{C}[i]$ and $\texttt{cand}[i]$ as the $i$-th sorted point.
+
+There are 2 cases for our 3 points: either the turn from $\texttt{C}[i-1] \rightarrow \texttt{C}[i] \rightarrow \texttt{cand}[i]$ is counterclockwise or not. If the turn is not counterclockwise, then we know that our current set of points cannot be convex, and thus we pop the top element off $\texttt{C}$ and proceed to check convexity of the next candidate point. Otherwise the turn is counterclockwise, and thus we include it in our hull.
+
+To understand why, consider those three points: the top two points in the stack, and our candidate point. Since our turn is counterclockwise, we are looking for such points that make a counterclockwise turn. It is easy to see that when a clockwise turn it is made, the polygon can no longer be convex.
+
+<Spoiler title="Illustration">
+
+![Illustration of Convex Hull](./assets/convexhullpts.jpg)
+
+</Spoiler>
+
+
+<Spoiler title="Worked Example">
+
+**Example**
+
+Find the convex hull of the points $(1,2),(2,3),(5,3),(3,2),(2,0)$
+
+Using the steps from our Graham Scan:
+
+1. Sorting the points by counterclockwise angle: Our leftmost point is $(1,2)$, so we then sort all the points by their counterclockwise angle. If the angles are the same, we can tiebreak by distance. Our resulting sorted points are $(2,0),(3,2),(5,3),(2,3)$ with initial pivot $(1,2)$.
+
+2. Maintaining the stack:
+
+Let's denote the $i$-th sorted point as $\texttt{pts}[i]$.
+
+Our initial stack elements are $(1,2),(2,0)$. Then, starting from $\texttt{pts}[2]$, we process each point.
+
+First, we check if the turn between $(1,2), (2,0)$, and $(3,2)$, is counterclockwise, which it is. Hence, the point $\texttt{pts}[2]$ is inserted on our stack.
+
+Our stack now is then $(1,2),(2,0),(3,2)$.
+
+Then, we check the turn $(2,0),(3,2),(5,3)$ which is clockwise. Hence, we pop the stack until the counterclockwise property can be satisfied. In this case, only $(5,3)$ is popped. Afterwards, we push $(2, 3)$ to our stack because it satisfies the counterclockwise property.
+
+Our stack now is then $(1,2),(2,0),(5,3)$.
+
+Then, we check whether the turn between $(2,0), (5,3), (2,3)$ is counterclockwise, which it is, so we push $(2,3)$ to our stack.
+
+Since we have processed all points, we have completed the cycle, and our convex hull is complete.
+
+Our stack now is then $(1,2),(2,0),(5,3),(2,3)$, which is the convex hull of all our points.
```

Comment:
Say "vertices of the convex hull" instead of "the convex hull" because the convex hull consists of all points within the border of the polygon bounded by these points.